syntax = "proto3";

service Borrow{
    rpc RequestLoan(BorrowerRequest) returns (BorrowerResponse);
    rpc RequestCheck(BankReturnCheckRequest) returns (BankReturnCheckResponse);
    rpc RequestInstallment(DeferredInstallmentRequest) returns (DeferredInstallmentResponse);

}

message BorrowerRequest {
    int64 Id = 1;
    int64 n_account = 2;
}

message BorrowerResponse {
    string message = 1;
}

message BankReturnCheckRequest {
    int64 Id = 1;
    int64 n_account = 2;
}

message BankReturnCheckResponse {
    bool message = 1;
}

message DeferredInstallmentRequest {
    int64 Id = 1;
    int64 n_account = 2;
}

message DeferredInstallmentResponse {
    bool message = 1;
}